Source/Core/Core/IOS/DI/DI.cpp

```diff
@@ -30,93 +34,618 @@ void DI::DoState(PointerWrap& p)
+    {
+      ERROR_LOG(IOS_DI,
+                "DVDLowOpenPartition with cert chain - not implemented, ignoring certs parameter");
+      DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    }
 
     const u64 partition_offset =
         static_cast<u64>(Memory::Read_U32(request.in_vectors[0].address + 4)) << 2;
-    const DiscIO::Partition partition(partition_offset);
-    DVDInterface::ChangePartition(partition);
+    ChangePartition(DiscIO::Partition(partition_offset));
 
-    INFO_LOG(IOS_DI, "DVDLowOpenPartition: partition_offset 0x%016" PRIx64, partition_offset);
+    INFO_LOG(IOS_DI, "DVDLowOpenPartition: partition_offset 0x%09" PRIx64, partition_offset);
 
     // Read TMD to the buffer
-    const IOS::ES::TMDReader tmd = DVDThread::GetTMD(partition);
+    const IOS::ES::TMDReader tmd = DVDThread::GetTMD(m_current_partition);
     const std::vector<u8>& raw_tmd = tmd.GetBytes();
     Memory::CopyToEmu(request.io_vectors[0].address, raw_tmd.data(), raw_tmd.size());
-    m_ios.GetES()->DIVerify(tmd, DVDThread::GetTicket(partition));
 
-    return_value = 1;
+    ReturnCode es_result = m_ios.GetES()->DIVerify(tmd, DVDThread::GetTicket(m_current_partition));
+    Memory::Write_U32(es_result, request.io_vectors[1].address);
+
+    return_value = SUCCESS;
     break;
   }
+  case DVDLowGetNoDiscOpenPartitionParams:
+    ERROR_LOG(IOS_DI, "DVDLowGetNoDiscOpenPartitionParams - dummied out");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    request.DumpUnknown(GetDeviceName(), LogTypes::IOS_DI);
+    break;
+  case DVDLowNoDiscOpenPartition:
+    ERROR_LOG(IOS_DI, "DVDLowNoDiscOpenPartition - dummied out");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    request.DumpUnknown(GetDeviceName(), LogTypes::IOS_DI);
+    break;
+  case DVDLowGetNoDiscBufferSizes:
+    ERROR_LOG(IOS_DI, "DVDLowGetNoDiscBufferSizes - dummied out");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    request.DumpUnknown(GetDeviceName(), LogTypes::IOS_DI);
+    break;
+  case DVDLowOpenPartitionWithTmdAndTicket:
+    ERROR_LOG(IOS_DI, "DVDLowOpenPartitionWithTmdAndTicket - not implemented");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    break;
+  case DVDLowOpenPartitionWithTmdAndTicketView:
+    ERROR_LOG(IOS_DI, "DVDLowOpenPartitionWithTmdAndTicketView - not implemented");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    break;
   default:
+    ERROR_LOG(IOS_DI, "Unknown ioctlv 0x%02x", request.request);
     request.DumpUnknown(GetDeviceName(), LogTypes::IOS_DI);
   }
   return GetDefaultReply(return_value);
 }
+
+void DI::ChangePartition(const DiscIO::Partition& partition)
+{
+  m_current_partition = partition;
+}
+
+const DiscIO::Partition& DI::GetCurrentPartition()
```

Comment:
```suggestion
DiscIO::Partition DI::GetCurrentPartition()
```

A partition simply consists of a u64 value. It's fine to return by value.